src/App.tsx

```diff
@@ -1,24 +1,193 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { SortBy } from './types/SortBy';
+import { addTodo, deleteTodo } from './api/todos';
+import { Todos } from './components/Todos';
+import { useGetTodos } from './hooks';
 
-const USER_ID = 0;
+const USER_ID = 11361;
 
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState<SortBy>(SortBy.all);
+  const [selectedTodo, setSelectedTodo] = useState<number[]>([]);
+  const [inputValue, setInputValue] = useState<string>('');
+  const [isLoadingTodo, setIsLoadingTodo] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const {
+    isLoading,
+    todos,
+    errorMessage,
+    todosNotCompleted,
+    handleError,
+    handleIsTodoDeleted,
+  } = useGetTodos(USER_ID, tempTodo);
+
+  const deleteOneTodo = async (todoId: number) => {
+    try {
+      await deleteTodo(USER_ID, todoId);
+    } catch (error) {
+      handleError('Unable to delete a todo');
+    }
+
+    handleIsTodoDeleted(true);
+    setSelectedTodo(prevSelectedTodo => [...prevSelectedTodo, todoId]);
+  };
+
+  const handleDeleteTodo = (value: number) => deleteOneTodo(value);
+
+  const handleKeyDown = async (
+    event: React.KeyboardEvent<HTMLInputElement>,
+  ) => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      if (!inputValue) {
+        handleError("Title can't be empty");
+      }
+
+      setIsLoadingTodo(true);
+
+      setTempTodo({
+        id: 0,
+        userId: 11361,
+        title: inputValue,
+        completed: false,
+      });
+
+      const todo = {
+        id: 0,
+        userId: 11361,
+        title: inputValue,
+        completed: false,
+      };
+
+      setInputValue('');
```

Comment:
This clear input value statement should be placed after the addTodo function. If the function throws an error the user will lose the input data.